integration-tests/contracts/ethereum_mercury_contracts.go

```diff
@@ -0,0 +1,224 @@
+package contracts
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts/ethereum/mercury/exchanger"
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts/ethereum/mercury/verifier"
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts/ethereum/mercury/verifier_proxy"
+)
+
+type Exchanger interface {
+	Address() string
+	CommitTrade(commitment [32]byte) error
+	ResolveTrade(encodedCommitment []byte) (string, error)
+	ResolveTradeWithReport(chainlinkBlob []byte, encodedCommitment []byte) (*types.Receipt, error)
+}
+
+type EthereumExchanger struct {
+	address   *common.Address
+	client    blockchain.EVMClient
+	exchanger *exchanger.Exchanger
+}
+
+func (v *EthereumExchanger) Address() string {
+	return v.address.Hex()
+}
+
+func (e *EthereumExchanger) CommitTrade(commitment [32]byte) error {
+	txOpts, err := e.client.TransactionOpts(e.client.GetDefaultWallet())
+	if err != nil {
+		return err
+	}
+	tx, err := e.exchanger.CommitTrade(txOpts, commitment)
+	if err != nil {
+		return err
+	}
+	return e.client.ProcessTransaction(tx)
+}
+
+func (e *EthereumExchanger) ResolveTrade(encodedCommitment []byte) (string, error) {
+	callOpts := &bind.CallOpts{
+		From:    common.HexToAddress(e.client.GetDefaultWallet().Address()),
+		Context: context.Background(),
+	}
+	data, err := e.exchanger.ResolveTrade(callOpts, encodedCommitment)
+	if err != nil {
+		return "", err
+	}
+	return data, nil
+}
+
+func (e *EthereumExchanger) ResolveTradeWithReport(chainlinkBlob []byte, encodedCommitment []byte) (*types.Receipt, error) {
+	txOpts, err := e.client.TransactionOpts(e.client.GetDefaultWallet())
+	txOpts.GasLimit = 1000000
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Need to perform err check first otherwise there is chance of txOpts returning as nil

```suggestion
      if err != nil {
		return nil, err
	}
	txOpts.GasLimit = 1000000
```

Reply:
@AnieeG sure, I'll fix it